### PR TITLE
Correct terminology

### LINKED ISSUE #16437: Do you assign values to variables or variables to values?
This issue is more about general terminology than the specific topic of this document (nullability). There are a few places in this document where the term "assign" is used like a value is being assigned to a variable:

> The variable can never be assigned the value null.

> These variables may be initialized with the default null value and may be assigned the value null in other code.

> Nonnullable: Null can't be assigned to variables of this type.

> Nullable: Null can be assigned to variables of this type.

However, there are other places in this document where it's like a variable is being assigned to a value:

> A reference can be null. No warnings are issued when a reference type is initialized to null, or later assigned to null.

> Oblivious: This is the pre-C# 8.0 state. Variables of this type can be dereferenced or assigned without warnings.

> All reference variables may be assigned to null.

I believe the correct way to use the term "assign" is as though values are assigned to variables. Consider [this other document](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/value-types):

> When you assign a new value to a variable of a value type, that value is copied. When you assign a new value to a variable of a reference type, the reference is copied, not the object itself.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: a690704d-da32-dca7-ea3e-66b84fb1d00d
* Version Independent ID: 11f314de-dd3a-362e-b2bb-d0a2133cfc7b
* Content: [Nullable reference types](https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references#nullable-contexts)
* Content Source: [docs/csharp/nullable-references.md](https://github.com/dotnet/docs/blob/master/docs/csharp/nullable-references.md)
* Product: **dotnet-csharp**
* Technology: **csharp-null-safety**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #16437

### CODE REVIEW

File: docs/csharp/nullable-references.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ C# 8.0 introduces **nullable reference types** and **non-nullable reference type
 
 This new feature provides significant benefits over the handling of reference variables in earlier versions of C# where the design intent couldn't be determined from the variable declaration. The compiler didn't provide safety against null reference exceptions for reference types:
 
-- **A reference can be null**. No warnings are issued when a reference type is initialized to null, or later assigned to null.
+- **A reference can be null**. No warnings are issued when a reference type is initialized to null, or null assigned to it later.
```
Comment (gewarren) [inline docs/csharp/nullable-references.md]:
```suggestion
- **A reference can be null**. No warning is issued when a reference type is initialized to null, or null is later assigned to it.
```

File: docs/csharp/nullable-references.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ Any reference type can have one of four *nullabilities*, which describes when wa
 
 - *Nonnullable*: Null can't be assigned to variables of this type. Variables of this type don't need to be null-checked before dereferencing.
 - *Nullable*: Null can be assigned to variables of this type. Dereferencing variables of this type without first checking for `null` causes a warning.
-- *Oblivious*: This is the pre-C# 8.0 state. Variables of this type can be dereferenced or assigned without warnings.
+- *Oblivious*: This is the pre-C# 8.0 state. Variables of this type can be dereferenced or be assigned without warnings.
```
Comment (gewarren) [inline docs/csharp/nullable-references.md]:
```suggestion
- *Oblivious*: This is the pre-C# 8.0 state. Variables of this type can be dereferenced or assigned to without warnings.
```

File: docs/csharp/nullable-references.md [hunk 3]
```diff
@@ -87,7 +87,7 @@ By default, nullable annotation and warning contexts are **disabled**. That mean
 The compiler uses the following rules in a disabled nullable annotation context:
 
 - You can't declare nullable references in a disabled context.
-- All reference variables may be assigned to null.
+- null may be assigned to all reference variables.
```
Comment (gewarren) [inline docs/csharp/nullable-references.md]:
```suggestion
- All reference variables may be assigned a value of null.
```

**Review by gewarren — CHANGES_REQUESTED** [review]
Thanks for fixing this @Youssef1313.

File: docs/csharp/nullable-references.md [hunk 4]
```diff
@@ -42,7 +42,7 @@ Any reference type can have one of four *nullabilities*, which describes when wa
 
 - *Nonnullable*: Null can't be assigned to variables of this type. Variables of this type don't need to be null-checked before dereferencing.
 - *Nullable*: Null can be assigned to variables of this type. Dereferencing variables of this type without first checking for `null` causes a warning.
-- *Oblivious*: This is the pre-C# 8.0 state. Variables of this type can be dereferenced or assigned without warnings.
+- *Oblivious*: This is the pre-C# 8.0 state. Variables of this type can be dereferenced or assigned to without warnings.
```
Comment (gewarren) [inline docs/csharp/nullable-references.md]: On second thought, I think the original text is more correct. A variable is assigned a value. A value is assigned to a variable. So I think that "variables...can be...assigned without warnings" is valid. Thoughts?

Comment (Youssef1313) [reply docs/csharp/nullable-references.md]: After reading the original text again, I think it depends on how it's interpreted.

Original text is:

> Variables of this type can be dereferenced or assigned without warnings.

The two interpretations with different meanings are:

> (Variables of this type can be (dereferenced or assigned) without warnings.

> (Variables of this type can ((be dereferenced) or (assigned)) without warnings.

I'm not sure if I can express what I mean well by that.

Comment (Youssef1313) [reply docs/csharp/nullable-references.md]: Had to read it several times 😆 .
Yes, it seems the original text is correct.

It says: "Variables can be assigned", which seems correct to me. I'm going to revert this change.

```suggestion
- *Oblivious*: This is the pre-C# 8.0 state. Variables of this type can be dereferenced or assigned without warnings.
```

**Review by gewarren — APPROVED** [review]
LG, thank you!

**Pull request merged**